stag-library-compiler/src/main/java/com/vimeo/stag/processor/generators/model/AnnotatedClass.java

```diff
@@ -51,16 +50,62 @@
     private final Element mElement;
 
     @NotNull
-    private final List<VariableElement> mMemberVariables;
+    private final Map<Element, TypeMirror> mMemberVariables;
 
-    @Nullable
-    private final TypeMirror mInheritedType;
-
-    public AnnotatedClass(@NotNull Element element, @NotNull List<VariableElement> members) {
+    AnnotatedClass(@NotNull Element element) {
         mType = element.asType();
         mElement = element;
-        mInheritedType = TypeUtils.getInheritedType(element);
-        mMemberVariables = new ArrayList<>(members);
+        TypeMirror inheritedType = TypeUtils.getInheritedType(element);
+        mMemberVariables = new HashMap<>();
+        for (Element enclosedElement : element.getEnclosedElements()) {
+            addToSupportedTypes(enclosedElement);
+        }
+
+        if (inheritedType != null) {
+            DebugLog.log(TAG, "\t\tInherited Type - " + inheritedType.toString());
+
+            AnnotatedClass genericInheritedType =
+                    SupportedTypesModel.getInstance().getSupportedType(inheritedType);
+
+            mMemberVariables.putAll(TypeUtils.getConcreteMembers(inheritedType, genericInheritedType.getElement(),
+                    genericInheritedType.getMemberVariables()));
+        }
+
+    }
+
+    private static void checkModifiers(VariableElement variableElement, Set<Modifier> modifiers) {
+        if (modifiers.contains(Modifier.FINAL)) {
+            if (!modifiers.contains(Modifier.STATIC)) {
+                throw new RuntimeException("Unable to access field \"" +
+                        variableElement.getSimpleName().toString() + "\" in class " +
+                        variableElement.getEnclosingElement().asType() +
+                        ", field must not be final.");
+            }
+        } else if (modifiers.contains(Modifier.PRIVATE)) {
+            throw new RuntimeException("Unable to access field \"" +
+                    variableElement.getSimpleName().toString() + "\" in class " +
+                    variableElement.getEnclosingElement().asType() +
+                    ", field must not be private.");
+        }
+    }
+
+    private void addToSupportedTypes(@NotNull Element element) {
+        if (element instanceof VariableElement) {
+            final VariableElement variableElement = (VariableElement) element;
+            Set<Modifier> modifiers = variableElement.getModifiers();
+            if ((!modifiers.contains(Modifier.FINAL) || !modifiers.contains(Modifier.STATIC)) && !modifiers.contains(Modifier.TRANSIENT)) {
```

Comment:
Yes. We'll update the readme doc with all the changes/features that have been integrated